docs/v8-migration-api.md

```diff
@@ -204,6 +237,24 @@ Check out the [v8-migration guide](v8-migration.md) for other user-level changes
   }
   ```
 
+- The second argument (`body`) of `TSModuleDeclaration` cannot be a `TSModuleDeclaration` anymore ([#16982](https://github.com/babel/babel/pull/16982))
+
+  __Migration__: Either create a `TSModuleBlock` body that exports the `TSModuleDeclaration`, or create a
+  single `TSModuleDeclaration` that has a `TSQualifiedName` as its name.
+
+  ```diff title="my-babel-codemod.js"
+    // Create `namespace X.Y {}`
+    t.tsModuleDeclaration(
+  +   t.tsQualifiedName(
+        t.identifier("X"),
+  -   t.tsModuleDeclaration(
+        t.identifier("Y"),
+  +   ),
+      t.tsModuleBlock([])
+  -   )
+    )
+  ```
```

Comment:
I would expect there are two examples here since we have mentioned two migration approaches. This example seems to be of the second one. Could you add an example for the first approach?